apm-agent-plugins/apm-spring-webmvc-plugin/src/main/java/co/elastic/apm/agent/spring/webmvc/ViewRenderInstrumentation.java

```diff
@@ -0,0 +1,143 @@
+    private static final String SPAN_ACTION = "render";
+    private static final String DISPATCHER_SERVLET_RENDER_METHOD = "View#render";
+    private static Map<String, String> subTypeCache = new ConcurrentHashMap<>();
+
+    @Override
+    public Class<?> getAdviceClass() {
+        return ViewRenderAdviceService.class;
+    }
+
+    public static class ViewRenderAdviceService {
+
+        @Advice.OnMethodEnter(suppress = Throwable.class)
+        public static void beforeExecute(@Advice.Local("span") @Nullable Span span,
+                                         @Advice.This @Nullable Object thiz) {
+            if (tracer == null || tracer.getActive() == null) {
+                return;
+            }
+            final TraceContextHolder<?> parent = tracer.getActive();
+
+            String className = thiz.getClass().getName();
+            span = parent.createSpan()
+                .withType(SPAN_TYPE)
+                .withSubtype(defineSubtype(className))
+                .withAction(SPAN_ACTION)
+                .withName(DISPATCHER_SERVLET_RENDER_METHOD);
+
+            if (thiz instanceof AbstractView) {
+                AbstractView view = (AbstractView) thiz;
+                span.appendToName(" ").appendToName(view.getBeanName());
+            }
+            span.activate();
+        }
+
+        @Advice.OnMethodExit(suppress = Throwable.class, onThrowable = Throwable.class)
+        public static void afterExecute(@Advice.Local("span") @Nullable Span span,
+                                        @Advice.Thrown @Nullable Throwable t) {
+            if (span != null) {
+                span.captureException(t)
+                    .deactivate()
+                    .end();
+            }
+        }
+
+        public static String defineSubtype(String className) {
+            switch (className) {
+                case "org.springframework.web.servlet.view.groovy.GroovyMarkupView":
+                    return "GroovyMarkup";
+                case "org.springframework.web.servlet.view.freemarker.FreeMarkerView":
+                    return "FreeMarker";
+                case "org.springframework.web.servlet.view.json.MappingJackson2JsonView":
+                    return "MappingJackson2Json";
+                case "de.neuland.jade4j.spring.view.JadeView":
+                    return "Jade";
+                case "org.springframework.web.servlet.view.InternalResourceView":
+                    return "InternalResource";
+                case "org.thymeleaf.spring4.view.ThymeleafView":
+                    return "Thymeleaf";
+                default:
+                    if (subTypeCache.containsKey(className)) {
+                        return subTypeCache.get(className);
+                    } else {
+                        String subType = className.substring(className.lastIndexOf('.') + 1, className.lastIndexOf("View"));
+                        subTypeCache.put(className, subType);
```

Comment:
```suggestion
                        if (subTypeCache.size() < 1000) {
                            subTypeCache.put(className, subType);
                        }
```
Just precaution from strange unknown scenarios